system-contracts/contracts/EvmInterpreter.yul

```diff
@@ -0,0 +1,1187 @@
+                mstore(DEBUG_SLOT_OFFSET(), 0x4A15830341869CAA1E99840C97043A1EA15D2444DA366EFFF5C43B4BEF299681)
+                $llvm_NoInline_llvm$_unoptimized()
+            }
+
+            function isSlotWarm(key) -> isWarm {
+                // TODO: Unhardcode this selector 0x482d2e74
+                mstore8(0, 0x48)
+                mstore8(1, 0x2d)
+                mstore8(2, 0x2e)
+                mstore8(3, 0x74)
+                mstore(4, key)
+
+                let success := call(gas(), EVM_GAS_MANAGER_CONTRACT(), 0, 0, 36, 0, 32)
+    
+                if iszero(success) {
+                    // This error should never happen
+                    revert(0, 0)
+                }
+    
+                isWarm := mload(0)
+            }
+
+            function warmSlot(key,currentValue) -> isWarm, originalValue {
+                // TODO: Unhardcode this selector 0xbdf78160
+                mstore8(0, 0xbd)
+                mstore8(1, 0xf7)
+                mstore8(2, 0x81)
+                mstore8(3, 0x60)
+                mstore(4, key)
+                mstore(36,currentValue)
+
+                let success := call(gas(), EVM_GAS_MANAGER_CONTRACT(), 0, 0, 68, 0, 64)
+    
+                if iszero(success) {
+                    // This error should never happen
+                    revert(0, 0)
+                }
+    
+                isWarm := mload(0)
+                originalValue := mload(32)
+            }
+
+            ////////////////////////////////////////////////////////////////
+            //                      FALLBACK
+            ////////////////////////////////////////////////////////////////
+
+            // TALK ABOUT THE DIFFERENCE BETWEEN VERBATIM AND DOING A STATIC CALL.
+            // IN SOLIDITY A STATIC CALL IS REPLACED BY A VERBATIM IF THE ADDRES IS ONE
+            // OF THE ONES IN THE SYSTEM CONTRACT LIST WHATEVER.
+            // IN YUL NO SUCH REPLACEMENTE TAKES PLACE, YOU NEED TO DO THE VERBATIM CALL
+            // MANUALLY.
+
+            let evmGasLeft, isStatic, isCallerEVM := consumeEvmFrame()
+
+            // TODO: Check if caller is not EVM and override evmGasLeft and isStatic with their
+            // appropriate values if so.
+
+            // First, copy the contract's bytecode to be executed into the `BYTECODE_OFFSET`
+            // segment of memory.
+            getDeployedBytecode()
+
+            // stack pointer - index to first stack element; empty stack = -1
+            let sp := sub(STACK_OFFSET(), 32)
+            // instruction pointer - index to next instruction. Not called pc because it's an
+            // actual yul/evm instruction.
+            let ip := add(BYTECODE_OFFSET(), 32)
+            let opcode
+
+            let returnOffset := MEM_OFFSET_INNER()
+            let returnLen := 0
+
+            for { } true { } {
+                opcode := readIP(ip)
+
+                ip := add(ip, 1)
+
+                switch opcode
```

Comment:
I believe YUL handling of switch statements is [very inefficient](https://github.com/ethereum/solidity/blob/6c7e686d86c33b3f329a6962728a1f1bed69d67a/libyul/backends/evm/EVMCodeTransform.cpp#L321). You may need to write this in [BALLS](https://github.com/Philogy/balls) instead.